### PR TITLE
Cleaning up <DMTCP_ROOT>/test/plugin

### PR DESCRIPTION
**Pull request by gc00**
Look at README, and one Makefile to see the pattern.
Three commits:
* README:  make clean tidy -> make clean
* Makefile:  dmtcp_NAMEhijack.so -> libdmtcp_NAME.so (and use word 'plugin library' in comments; not 'hijack library')
* Makefile:  LINKER = ${CC}    [ and the .so target uses ${LINKER} ... ]

QUESTION:  Is there a convention for the macro name LINKER (used to create a .so file from .o files)?

### CODE REVIEW

**Comment by karya0** [comment]
We use `CXXLINK` as the macro name for `LINKER` that you used above. http://www.gnu.org/software/automake/manual/html_node/How-the-Linker-is-Chosen.html

File: test/plugin/applic-delayed-ckpt/Makefile [hunk 1]
```diff
@@ -17,6 +17,7 @@ DMTCP_INCLUDE=${DMTCP_ROOT}/include
 
 CFLAGS += -I${DMTCP_INCLUDE} -DDMTCP -fPIC -c -g
 CXXFLAGS += -I${DMTCP_INCLUDE} -DDMTCP -fPIC -c -g
+LINKER = ${CC}
```
Comment (karya0) [inline test/plugin/applic-delayed-ckpt/Makefile]: s/LINKER/LINK ?

http://www.gnu.org/software/automake/manual/html_node/How-the-Linker-is-Chosen.html

File: test/plugin/applic-initiated-ckpt/Makefile [hunk 2]
```diff
@@ -17,6 +17,7 @@ DMTCP_INCLUDE=${DMTCP_ROOT}/include
 
 CFLAGS += -I${DMTCP_INCLUDE} -DDMTCP -fPIC -c -g
 CXXFLAGS += -I${DMTCP_INCLUDE} -DDMTCP -fPIC -c -g
+LINKER = ${CC}
```
Comment (karya0) [inline test/plugin/applic-initiated-ckpt/Makefile]: Same as above.

File: test/plugin/applic-initiated-ckpt/Makefile [hunk 3]
```diff
@@ -44,7 +45,7 @@ check: ${LIBNAME}.so applic
 
 # We link the library using C++ for compatibility with the main libdmtcp.so
 ${LIBNAME}.so: ${LIBOBJS}
-	${CXX} -shared -fPIC -o $@ $^
+	${LINKER} -shared -fPIC -o $@ $^
```
Comment (karya0) [inline test/plugin/applic-initiated-ckpt/Makefile]: ditto

File: test/plugin/example-db/Makefile [hunk 4]
```diff
@@ -17,6 +17,7 @@ DMTCP_INCLUDE=${DMTCP_ROOT}/include
 
 CFLAGS += -I${DMTCP_INCLUDE} -DDMTCP -fPIC -c -g
 CXXFLAGS += -I${DMTCP_INCLUDE} -DDMTCP -fPIC -c -g
+LINKER = ${CC}
```
Comment (karya0) [inline test/plugin/example-db/Makefile]: same as above

**Comment by karya0** [comment]
BTW, we use the following definition for `CXXLINK`:
 
```CXXLINK = $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@                                    
```

Let's use a similar one for `LINK`.

File: test/plugin/applic-delayed-ckpt/Makefile (hunk 1 above)
Comment (gc00) [inline test/plugin/applic-delayed-ckpt/Makefile]: Agreed (LINKER -> LINK).  Thanks.  I was looking for the right convention.

**Comment by gc00** [comment]
I changed LINKER -> LINK.  (Nice.  I was looking for the conventional name.)
As for `CXXLINK = $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ `,
the rest of that Makefile does things like:  `CFLAGS += -I${DMTCP_INCLUDE} -DDMTCP -fPIC -c -g`
followed by:  `${CC} ${CFLAGS} -o $@ $<`
In that spirit, I could define `LDFLAGS = -shared -fPIC`
and follow it by: ` ${LINK} ${LDFLAGS} -o $@ $^`

Whatever we do, this is intended as an easy example, readable instantly by a novice.  That means, keeping the recursion of macro expansion only one level deep.  Note that `LINK = ... ${LDFLAGS}` followed by `LDFLAGS = ...` would be two levels deep.

My first thought would be to leave it as is, but I don't mind introducing `LDFLAGS` if you prefer that.  Tell me what you'd prefer.

**Comment by karya0** [comment]
Let's keep it simple for this case. For new real plugins, we want to follow the same convention as with existing plugins.

**Comment by karya0** [comment]
If nothing else needs to be done, this PR can be merged.

**Comment by gc00** [comment]
Done. :-)

**Pull request closed without merging**